### PR TITLE
Dashboard: Making a dashboard editable does not allow resizing of panels

### PR DESCRIPTION
**Pull request by polibb**
**What is this feature?**

When `dashboard.meta.canEdit`/`canSave` is updated ([here](https://github.com/grafana/grafana/blob/polibb/rerender-dashboard-grid-when-editable-set/public/app/features/dashboard/components/DashboardSettings/DashboardSettings.tsx#L207)) the `DashboardGrid` should be rerendered in order to rerender `ReactGridLayout` with correct `isDraggable={true}` and `isResizable={true}` props.
It does not rerender because the dashboard prop defies the immutability principle and the object is the same. So, as the quickest/cleanest current solution, we can pass the `dashboard.meta.canEdit` value directly as it is not an object. 

**Why do we need this feature?**

When a read-only dashboard is set to "Make Editable" from its settings, the user can go back to the dashboard. The dashboard should now be editable and for the most part it is, except the Grid layout does not re-render, so it does not allow the resizing of panels as it should. 

**Which issue(s) does this PR fix?**:

Escalation https://github.com/grafana/support-escalations/issues/4493. 

Fixes #

**Special notes for your reviewer**:

### CODE REVIEW

File: public/app/features/dashboard/containers/DashboardPage.tsx [hunk 1]
```diff
@@ -392,7 +392,12 @@ export class UnthemedDashboardPage extends PureComponent<Props, State> {
             </section>
           )}
 
-          <DashboardGrid dashboard={dashboard} viewPanel={viewPanel} editPanel={editPanel} />
+          <DashboardGrid
+            dashboard={dashboard}
+            isEditable={!!dashboard.meta.canEdit}
```
Comment (ivanortegaalba) [inline public/app/features/dashboard/containers/DashboardPage.tsx:397]: Probably you can use the model-getter to make sure we can abstract the logic behind editing a dashboard.
```suggestion
            isEditable={dashboard.canEditDashboard()}
```

**Review by ivanortegaalba — CHANGES_REQUESTED** [review]
I'm not 100% sure, but I think we need to use the model.canEditDashboard() to also check the `canMakeEditable` boolean in the dashboard model
```js
canEditDashboard() {
  return Boolean(this.meta.canEdit || this.meta.canMakeEditable);
}
```

**Comment by polibb** [comment]
> I'm not 100% sure, but I think we need to use the model.canEditDashboard() to also check the canMakeEditable boolean in the dashboard model

> ```
> canEditDashboard() {
>   return Boolean(this.meta.canEdit || this.meta.canMakeEditable);
> }
> ```

@ivanortegaalba that's a good suggestion - to use the model's getters, but in this case it would not work because when a dashboard is read-only and it is opened, the `canMakeEditable` prop of the dashboard model is `true` because you can make it editable in the settings. 
image

**Review by ivanortegaalba — APPROVED** [review]
👍 After the double check, you're right, we cannot rely on that getter. LTGM

**Pull request merged**